src/main/java/core/basesyntax/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void updateDataBase(FruitTransaction transaction) {
+        if (FruitStorage.fruitStorage.get(transaction.getFruit()) == null) {
+            throw new RuntimeException("First you need to enter the balance of "
+                    + transaction.getFruit() + " under the code 'b'");
+        }
+        for (Map.Entry<String, Integer> entry : FruitStorage.fruitStorage.entrySet()) {
+            if (transaction.getFruit().equals(entry.getKey())) {
+                entry.setValue(entry.getValue() + transaction.getQuantity());
+            }
+        }
```

Comment:
Instead of iterating over the entire map, you can directly update the balance using the key. This can be done by checking if the fruit exists and then directly updating the value: `FruitStorage.fruitStorage.put(transaction.getFruit(), currentBalance + transaction.getQuantity());` This approach is more efficient and concise.